# Week 3 Assessment

This Jupyter Notebook is auxillary to the following assessment in this week.  To complete this assessment, you will complete the 5 questions outlined in this document and use the output from the Python cells to answer them.

Run the following cell to initialize your environment and begin the assessment.

In [1]:
import numpy as np
import statsmodels.api as sm
import pandas as pd  

url = "nhanes_2015_2016.csv"
da = pd.read_csv(url)

# Drop unused columns, drop rows with any missing values.
vars = ["BPXSY1", "RIDAGEYR", "RIAGENDR", "RIDRETH1", "DMDEDUC2", "BMXBMI",
        "SMQ020", "SDMVSTRA", "SDMVPSU"]
da = da[vars].dropna()

da["group"] = 10*da.SDMVSTRA + da.SDMVPSU

da["smq"] = da.SMQ020.replace({2: 0, 7: np.nan, 9: np.nan})

np.random.seed(123)

#### Question 1: What is clustered data? (You'll answer this question within the quiz that follows this notebook)

#### Answer:
Clustered data refers to observations that are naturally grouped together, meaning that measurements within the same cluster may be more similar to each other than to measurements in different clusters. This dependency can arise due to common characteristics shared by subjects within a cluster (e.g., individuals in the same household, students in the same school, or patients in the same hospital). Failing to account for clustering can lead to incorrect statistical inferences.


#### Question 2: (You'll answer this question within the quiz that follows this notebook)

Utilize the following output for this question:

In [2]:
for v in ["BPXSY1", "SDMVSTRA", "RIDAGEYR", "BMXBMI", "smq"]:
    model = sm.GEE.from_formula(v + " ~ 1", groups="group",
           cov_struct=sm.cov_struct.Exchangeable(), data=da)
    result = model.fit()
    print(v, result.cov_struct.summary())

BPXSY1 The correlation between two observations in the same cluster is 0.030
SDMVSTRA The correlation between two observations in the same cluster is 0.959
RIDAGEYR The correlation between two observations in the same cluster is 0.035
BMXBMI The correlation between two observations in the same cluster is 0.039
smq The correlation between two observations in the same cluster is 0.026


Which of the listed features has the highest correlation between two observations in the same cluster? 

Answer:

From the output:

Variable	Correlation within clusters
BPXSY1	0.030
SDMVSTRA	0.959
RIDAGEYR	0.035
BMXBMI	0.039
smq	0.026

SDMVSTRA has the highest correlation (0.959) between two observations in the same cluster, indicating that it is a strongly clustered variable.

#### Question 3: (You'll answer this question within the quiz that follows this notebook)

What is true about multiple linear regression and marginal linear models when dependence is present in data?

Answer:
When dependence is present in data, multiple linear regression assumes that observations are independent, which can lead to underestimated standard errors and incorrect statistical inferences. On the other hand, marginal linear models, such as Generalized Estimating Equations (GEE), account for within-cluster correlation, leading to more accurate and robust inference.


#### Question 4: (You'll answer this question within the quiz that follows this notebook)

Multilevel models are expressed in terms of _____.

Answer:
Multilevel models are expressed in terms of fixed effects (population-level effects) and random effects (group-specific deviations). These models explicitly model hierarchical data structures, where observations are nested within groups.


#### Question 5: (You'll answer this question within the quiz that follows this notebook)

Which of the following is NOT true regarding reasons why we fit marginal models?

Answer:
Marginal models (such as GEE) are typically used when the focus is on population-level effects rather than individual or cluster-specific effects.

A statement that is not true regarding marginal models could be:
“Marginal models provide cluster-specific predictions.”
This is false because marginal models estimate population-averaged effects rather than subject-specific effects (which are estimated by multilevel models).